### PR TITLE
Low: controld: Update for OCF 1.1

### PR DESCRIPTION
**Pull request by nrwahl2**

### CODE REVIEW

**Review by kgaillot — COMMENTED** [review]
Looks good so far. Also need to update the validate command to do the checks other than globally-unique only on level 10 (since they only apply to the node running the resource).

**Comment by nrwahl2** [comment]
> do the checks other than globally-unique only on level 10

Updated the OCF 1.1 commit accordingly. Also added one more commit on top to make the globally-unique check a bit more robust.

File: extra/resources/controld.in [hunk 1]
```diff
@@ -64,20 +64,23 @@ In most cases, it should be run as an anonymous clone.
...
+<parameter name="daemon" unique-group="daemon">
 <longdesc lang="en">
 The daemon to start - supports gfs_controld and dlm_controld
 </longdesc>
 <shortdesc lang="en">The daemon to start</shortdesc>
-<content type="string" default="dlm_controld" />
+<content type="select" default="dlm_controld">
+  <option value="gfs_controld" />
+  <option value="dlm_controld" />
+</content>
```
Comment (kgaillot) [inline extra/resources/controld.in:83]: Looking at this again, I don't think we restricted the daemon name before -- these were just the possible defaults

Theoretically someone could have a custom daemon under a different name

File: extra/resources/controld.in [hunk 2]
```diff
@@ -260,17 +263,19 @@ controld_monitor() {
...
             exit $OCF_ERR_CONFIGURED
             ;;
     esac
 
-    [ -d "@runstatedir@/cluster" ] || mkdir "@runstatedir@/cluster"
+    # Host-specific checks
+    if [ "$OCF_CHECK_LEVEL" = "10" ]; then
+        check_binary killall
+        check_binary "${OCF_RESKEY_daemon}"
+        [ -d "@runstatedir@/cluster" ] || mkdir "@runstatedir@/cluster"
```
Comment (kgaillot) [inline extra/resources/controld.in:277]: Ouch, I just realized this was a little hacky before -- not only did it validate, but it created this directory if needed. start/monitor call validate (without a check level). So we still need to do this, but it doesn't belong in the validate function

**Review by kgaillot — COMMENTED** [review]
Noticed a couple more things

File: extra/resources/controld.in (hunk 1 above)
Comment (nrwahl2) [reply extra/resources/controld.in:83]: Good catch

File: extra/resources/controld.in (hunk 2 above)
Comment (nrwahl2) [reply extra/resources/controld.in]: Ahhh yep.

**Comment by nrwahl2** [comment]
Replaced last two commits to address review.
* Reverted `daemon` content type to `string` since a user could specify a custom daemon if they wanted.
* Moved `@runstatedir@/cluster` check and creation from `controld_validate()` to `controld_start()`.

**Comment by nrwahl2** [comment]
The OCF 1.1 update probably should have had "Feature" in the header. May want to write it down somewhere for Changelog inclusion.

**Pull request merged**